pkg/signing/signing.go

```diff
@@ -0,0 +1,130 @@
+package signing
+
+import (
+	"bufio"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"golang.org/x/crypto/sha3"
+)
+
+// Signer allows you to sign a big stream of bytes by calling Sum multiple times, then Sign.
+type Signer struct {
+	state      crypto.KeccakState
+	privateKey *ecdsa.PrivateKey
+}
+
+// HexToECDSA parses a hex-encoded secp256k1 private key string to an ECDSA
+// private key.
+func HexToECDSA(hexKey string) (*ecdsa.PrivateKey, error) {
+	return crypto.HexToECDSA(hexKey)
+}
+
+// FileToECDSA parses a file path to a hex-encoded secp256k1 private key to an
+// ECDSA private key.
+func FileToECDSA(hexPath string) (*ecdsa.PrivateKey, error) {
```

Comment:
added a new method to create an ecdsa key from a private key that's located at a file